packages/miniflare/src/http/fetch.ts

```diff
@@ -99,9 +92,85 @@ export type DispatchFetch = (
 ) => Promise<Response>;
 
 export type AnyHeaders = http.IncomingHttpHeaders | string[];
-function addHeader(/* mut */ headers: AnyHeaders, key: string, value: string) {
-	if (Array.isArray(headers)) headers.push(key, value);
-	else headers[key] = value;
+function addHeader(
+	/* mut */ headers: undici.Headers,
+	key: string,
+	value: string
+) {
+	headers.set(key, value);
+}
+
+function isArray(headers: UndiciHeaders): headers is string[] {
+	return Array.isArray(headers);
+}
+
+function isIterable(
+	headers: UndiciHeaders
+): headers is Iterable<[string, string | string[] | undefined]> {
+	return Symbol.iterator in Object(headers);
+}
+
+// See https://github.com/nodejs/undici/blob/main/docs/docs/api/Dispatcher.md?plain=1#L1151 for documentation
+function convertUndiciHeadersToStandard(
+	headers: NonNullable<UndiciHeaders>
+): undici.Headers {
```

Comment:
Further down I see code that looks like:

```ts
new Headers(init.headers)
```

Why can't we use that here?

Reply:
The allowed values for `UndiciHeaders` are wider than the `Header` constructor can take (in particular, `string[]`)

Reply:
The allowed values for UndiciHeaders are wider than the Header constructor can take (in particular, string[])